src/main/java/org/motechproject/prevac/service/impl/PrimeVaccinationScheduleServiceImpl.java

```diff
@@ -55,47 +55,78 @@
 
     @Override
     public PrimeVaccinationScheduleDto createOrUpdateWithDto(PrimeVaccinationScheduleDto dto, Boolean ignoreLimitation) {
-        Visit primeDetails = visitBookingDetailsDataService.findById(dto.getVisitBookingDetailsId());
-        Visit screeningDetails = getScreeningDetails(primeDetails);
+        validateDate(dto);
 
-        if (primeDetails == null || screeningDetails == null) {
-            throw new IllegalArgumentException("Cannot save, because details for Visit not found");
+        Visit primeVisit = visitBookingDetailsDataService.findById(dto.getVisitBookingDetailsId());
+        // We have an update
+        if (primeVisit != null) {
+            Visit screeningVisit = getScreeningDetails(primeVisit);
+            if (screeningVisit == null) {
+                throw new IllegalArgumentException("Cannot save, because details for Visit not found");
```

Comment:
Change the exception message to "Cannot save, because Screening for Visit not found"